internal/config/main.go

```diff
@@ -61,9 +62,10 @@ func New(opts ...Option) Config {
 		targetAllocatorConfigMapEntry: defaultTargetAllocatorConfigMapEntry,
 		logger:                        logf.Log.WithName("config"),
 		openshiftRoutes:               newOpenShiftRoutesWrapper(),
+		hpaVersion:                    newHPAVersionWrapper(),
 		version:                       version.Get(),
-		autoscalingVersion:            autodetect.DefaultAutoscalingVersion,
 		onOpenShiftRoutesChange:       newOnChange(),
+		onHPAVersionChange:            newOnChange(),
```

Comment:
Do we need to register a callback for this function?

Reply:
or actually, do we need to call/use `WithOnHPAVersionChangeCallback` set up in `options.go`?